x-pack/platform/plugins/shared/triggers_actions_ui/public/application/hooks/use_rule_template.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+
+import { useKibana } from '../../common/lib/kibana';
+import { loadRuleTemplate } from '../lib/rule_template_api/get_rule_template';
+
+export interface UseRuleTemplateProps {
+  templateId?: string;
+  enabled: boolean;
+}
+
+export function useRuleTemplate(props: UseRuleTemplateProps) {
+  const { enabled, templateId } = props;
+  const { http } = useKibana().services;
+
+  const queryFn = () => {
+    if (!templateId) {
+      throw new Error('templateId not defined');
+    }
+    return loadRuleTemplate({ http, templateId });
+  };
+
+  const { data, isFetching, isError, isLoadingError, isLoading } = useQuery({
+    queryKey: ['getQueryDelaySettings'],
+    queryFn,
+    enabled,
```

Comment:
```suggestion
    enabled,
    networkMode: 'always',
```

I was testing out this functionality, and this request was not firing for me. Looking into it a bit this due to < v5 useQuery uses what seems like an unreliable API (navigator.onLine: boolean) to determine the network status of the browser that can get stuck as false

https://github.com/TanStack/query/discussions/4753